src/Sulu/Bundle/MediaBundle/Serializer/Subscriber/MediaPermissionsSubscriber.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Sulu\Bundle\MediaBundle\Serializer\Subscriber;
+
+use JMS\Serializer\EventDispatcher\EventSubscriberInterface;
+use JMS\Serializer\EventDispatcher\ObjectEvent;
+use JMS\Serializer\Metadata\StaticPropertyMetadata;
+use JMS\Serializer\Visitor\SerializationVisitorInterface;
+use Sulu\Bundle\MediaBundle\Entity\Media;
+use Sulu\Component\Rest\ApiWrapper;
+use Sulu\Component\Security\Authorization\AccessControl\AccessControlManagerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class MediaPermissionsSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var AccessControlManagerInterface
+     */
+    private $accessControlManager;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    public function __construct(
+        AccessControlManagerInterface $accessControlManagerInterface,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->accessControlManager = $accessControlManagerInterface;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            ['event' => 'serializer.post_serialize', 'method' => 'onPostSerialize'],
+        ];
+    }
+
+    public function onPostSerialize(ObjectEvent $event)
+    {
+        $object = $event->getObject();
+        /** @var SerializationVisitorInterface $visitor */
+        $visitor = $event->getVisitor();
+
+        // FIXME This should be removed, once all entities are restructured not using the ApiWrapper, possible BC break
```

Comment:
@wachterjohannes looks good but we should do this only for a specific group else we will maybe change accidently Website API responses.

Reply:
the subscriber is only registered in admin ad we do it for the "normal" `Sulu\Component\Security\Serializer\Subscriber\SecuredEntitySubscriber`

Reply:
We should not build context based services now, as the context will be removed in the future in favor of the single kernel setup.

Reply:
@alexander-schranz there is currently no `group` in the MediaController therefor we should keep it like the "original" service. everything else would be a BC Break and i would like to rebase that on 2.2